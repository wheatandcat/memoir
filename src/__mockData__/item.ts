import { ItemQuery } from 'queries/api/index';
type Item = NonNullable<ItemQuery['item']>;

export const item = (option?: Partial<Item>): Item => ({
  id: '1',
  title: '買い物',
  categoryID: 9,
  date: '2021-01-01T00:00:00+09:00',
  like: false,
  dislike: false,
  createdAt: '2021-01-01T00:00:00+09:00',
  updatedAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const items = (): Item[] => [
  item(),
  item({
    id: '2',
    title: '本を読む',
    categoryID: 2,
    date: '',
    dislike: false,
    like: false,
  }),
];
