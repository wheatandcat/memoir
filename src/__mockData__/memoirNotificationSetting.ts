import { State as MemoirNotificationSettingState } from 'hooks/useMemoirNotificationSetting';

export const memoirNotificationSetting = (
  option?: Partial<MemoirNotificationSettingState>
): MemoirNotificationSettingState => ({
  dayOfWeek: 0,
  hours: 0,
  minutes: 0,
  notification: false,
  ...option,
});
