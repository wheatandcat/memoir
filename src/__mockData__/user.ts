import { User } from 'queries/api/index';

export const user = (option?: Partial<User>): User => ({
  id: '1',
  displayName: 'test',
  image: '',
  updatedAt: '2021-01-01T00:00:00+09:00',
  createdAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const users = (): User[] => [
  {
    id: 'test1',
    displayName: 'suzuki1',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test2',
    displayName: 'suzuki2',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test3',
    displayName: 'suzuki4',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test4',
    displayName: 'suzuki4',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test5',
    displayName: 'suzuki5',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
];
