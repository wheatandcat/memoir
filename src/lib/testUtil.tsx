import React from 'react';
import { GraphQLHandler, GraphQLRequest } from 'msw';
import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
} from '@apollo/client';
import { render } from '@testing-library/react-native';
import { server } from 'mocks/server';
import fetch from 'cross-fetch';

const link = createHttpLink({
  uri: 'http://localhost:8080/query',
  fetch,
  credentials: 'same-origin',
});

const client = new ApolloClient({
  link,
  uri: 'http://localhost:8080/query',
  cache: new InMemoryCache(),
});

export const testRenderer =
  (children: React.ReactNode) =>
  (responseOverride?: GraphQLHandler<GraphQLRequest<never>>) => {
    if (responseOverride) {
      server.use(responseOverride);
    }
    return render(<ApolloProvider client={client}>{children}</ApolloProvider>);
  };
