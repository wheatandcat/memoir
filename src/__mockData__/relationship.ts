import type { RelationshipsQuery } from "queries/api/index";
import { user } from "./user";

export type Relationship = NonNullable<
  EdgesNode<RelationshipsQuery["relationships"]>
>;

export const relationship = (option?: Partial<Relationship>): Relationship => ({
  id: "1",
  followedId: "test",
  followerId: "test",
  user: user(),
  createdAt: "2021-01-01T00:00:00+09:00",
  updatedAt: "2021-01-01T00:00:00+09:00",
  ...option,
});

export const relationships = (): Relationship[] => [
  relationship(),
  relationship({
    id: "2",
    user: user(),
  }),
  relationship({
    id: "3",
    user: user(),
  }),
  relationship({
    id: "4",
    user: user(),
  }),
];
