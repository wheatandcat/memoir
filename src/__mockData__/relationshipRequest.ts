import { RelationshipRequestsQuery } from 'queries/api/index';
import { user } from './user';

export type RelationshipRequest = NonNullable<
  EdgesNode<RelationshipRequestsQuery['relationshipRequests']>
>;

export const relationshipRequest = (
  option?: Partial<RelationshipRequest>
): RelationshipRequest => ({
  id: '1',
  followedId: 'test',
  followerId: 'test',
  user: user(),
  createdAt: '2021-01-01T00:00:00+09:00',
  updatedAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const relationshipRequests = (): RelationshipRequest[] => [
  relationshipRequest(),
  relationshipRequest({
    id: '2',
    user: user(),
  }),
  relationshipRequest({
    id: '3',
    user: user(),
  }),
  relationshipRequest({
    id: '4',
    user: user(),
  }),
];
