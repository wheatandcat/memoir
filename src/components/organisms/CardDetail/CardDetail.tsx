import React, { memo, useCallback } from 'react';
import {
  StyleSheet,
  useWindowDimensions,
  ViewStyle,
  Alert,
} from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import Image from 'components/atoms/Image';
import Category from 'components/atoms/Category';
import theme from 'config/theme';
import dayjs from 'lib/dayjs';
import setting from 'components/atoms/Category/setting';
import { categoryBorderStyle } from 'lib/category';
import Menu, { Item as MenuItem } from 'components/organisms/Menu/Menu';

export type Props = {
  title: string;
  date: string;
  categoryID: number;
  like: boolean;
  dislike: boolean;
  onOpenUpdateItem: () => void;
  onDeleteItem: () => void;
};

const CardDetail: React.FC<Props> = (props) => {
  const windowWidth = useWindowDimensions().width;

  const titleStyle = {
    width: windowWidth - 60,
  };

  const category = setting().icon.find(
    (v) => v.id === props.categoryID
  )?.category;
  const categoryStyle: ViewStyle[] = [
    styles.root,
    categoryBorderStyle(category || 0),
  ];

  const menuItem = useCallback(
    (): MenuItem[] => [
      {
        text: '削除',
        color: 'error',
        onPress: (callback?: () => void) => {
          Alert.alert(
            '削除しますか？',
            '',
            [
              {
                text: 'キャンセル',
                style: 'cancel',
              },
              {
                text: '削除する',
                onPress: () => {
                  props.onDeleteItem();
                  callback?.();
                },
              },
            ],
            {
              cancelable: true,
            }
          );
        },
        removeMenu: false,
      },
      {
        text: '編集',
        color: 'secondary',
        onPress: () => {
          props.onOpenUpdateItem();
        },
        removeMenu: true,
      },
    ],
    [props]
  );

  return (
    <View style={categoryStyle}>
      <View style={styles.header}>
        <View>
          <Text>{dayjs(props.date).format('YYYY.MM.DD / ddd')}</Text>
        </View>
        <Menu items={menuItem()} />
      </View>
      <View style={styles.icon}>
        <Category categoryID={props.categoryID} />
        {props.like && (
          <View px={3}>
            <Image
              source={require('../../../img/icon/icon_like.png')}
              width={60}
              height={60}
              resizeMode="contain"
              testID="like"
            />
          </View>
        )}
        {props.dislike && (
          <View px={3}>
            <Image
              source={require('../../../img/icon/icon_dislike.png')}
              width={60}
              height={60}
              resizeMode="contain"
              testID="dislike"
            />
          </View>
        )}
      </View>
      <View style={[styles.title, titleStyle]}>
        <Text lineHeight={25}>{props.title}</Text>
      </View>
      <View style={styles.footer} />
    </View>
  );
};

export default memo(CardDetail);

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.light,
    borderLeftWidth: 5,
    paddingHorizontal: theme().space(3),
    paddingVertical: theme().space(2),
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    zIndex: 1,
  },
  title: {
    paddingBottom: theme().space(4),
  },
  icon: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: theme().space(4),
  },
  footer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});
