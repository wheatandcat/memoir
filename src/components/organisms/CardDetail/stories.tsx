import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { StyleSheet } from 'react-native';
import View from 'components/atoms/View';
import theme from 'config/theme';
import { mockFn } from 'storyBookUtils/index';
import CardDetail from './CardDetail';

storiesOf('organisms/CardDetail', module).add('CardDetail', () => (
  <View p={3} style={styles.root}>
    <CardDetail
      date="2021-02-21"
      categoryID={1}
      title="とても長いタイトルの本を読んだことがあって次もまた長いタイトルの本を読む"
      onOpenUpdateItem={mockFn('onOpenUpdateItem')}
      onDeleteItem={mockFn('onDeleteItem')}
    />
  </View>
));

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
  },
});
