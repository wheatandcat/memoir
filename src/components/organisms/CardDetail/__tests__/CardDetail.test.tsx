import React from 'react';
import { testRenderer } from 'lib/testUtil';
import { screen } from '@testing-library/react-native';
import { item } from '__mockData__/item';
import CardDetail, { Props } from '../CardDetail';

const propsData = (): Props => ({
  title: item().title,
  date: item().date,
  categoryID: item().categoryID,
  like: item().like,
  dislike: item().dislike,
  onOpenUpdateItem: jest.fn(),
  onDeleteItem: jest.fn(),
});

describe('components/organisms/CardDetail/CardDetail.tsx', () => {
  it('正常にrenderすること', () => {
    testRenderer(<CardDetail {...propsData()} />)();
    expect(screen.findAllByText('')).toBeTruthy();
  });
});
