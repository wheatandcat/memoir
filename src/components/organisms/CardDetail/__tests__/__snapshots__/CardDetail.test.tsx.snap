// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/CardDetail/CardDetail.tsx 正常にrenderすること 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderLeftWidth": 5,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      Object {
        "borderLeftColor": "#FFE600",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "zIndex": 1,
      }
    }
  >
    <View>
      <Text
        fontFamily="RobotoCondensed-Bold"
        variants="body"
      >
        2021.01.01 / 金
      </Text>
    </View>
    <Memo(Menu)
      items={
        Array [
          Object {
            "color": "error",
            "onPress": [Function],
            "removeMenu": false,
            "text": "削除",
          },
          Object {
            "color": "secondary",
            "onPress": [Function],
            "removeMenu": true,
            "testID": "edit",
            "text": "編集",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 32,
      }
    }
  >
    <Memo(Category)
      categoryID={9}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 32,
        },
        Object {
          "width": 690,
        },
      ]
    }
  >
    <Text
      fontFamily="RobotoCondensed-Bold"
      lineHeight={25}
      variants="body"
    >
      買い物
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  />
</View>
`;
